persistence/sql/persister_oauth2.go

```diff
@@ -483,10 +536,25 @@ func (p *Persister) RevokeRefreshToken(ctx context.Context, id string) (err erro
 	return p.deactivateSessionByRequestID(ctx, id, sqlTableRefresh)
 }
 
-func (p *Persister) RevokeRefreshTokenMaybeGracePeriod(ctx context.Context, id string, _ string) (err error) {
+func (p *Persister) RevokeRefreshTokenMaybeGracePeriod(ctx context.Context, id string, signature string) (err error) {
 	ctx, span := p.r.Tracer(ctx).Tracer().Start(ctx, "persistence.sql.RevokeRefreshTokenMaybeGracePeriod")
 	defer otelx.End(span, &err)
-	return p.deactivateSessionByRequestID(ctx, id, sqlTableRefresh)
+
+	gracePeriod := p.config.RefreshTokenRotationGracePeriod(ctx)
+	if gracePeriod <= 0 {
+		return p.deactivateSessionByRequestID(ctx, id, sqlTableRefresh)
+	}
+
+	/* #nosec G201 table is static */
+	return sqlcon.HandleError(
+		p.Connection(ctx).
+			RawQuery(
+				fmt.Sprintf("UPDATE %s SET active=false, first_used_at = CURRENT_TIMESTAMP WHERE request_id=? AND nid = ? AND active=true", OAuth2RequestSQL{Table: sqlTableRefresh}.TableName()),
+				id,
+				p.NetworkID(ctx),
+			).
+			Exec(),
+	)
```

Comment:
Do we need the branching here? I would prefer to just set the `first_used_at` column every time, even if the grace period is not defined. It will be handled correctly in `GetRefreshTokenSession` regardless.
The way it is right now, enabling the grace period would only affect new refresh requests.
```suggestion
	/* #nosec G201 table is static */
	return sqlcon.HandleError(
		p.Connection(ctx).
			RawQuery(
				fmt.Sprintf("UPDATE %s SET active=false, first_used_at = CURRENT_TIMESTAMP WHERE request_id=? AND nid = ? AND active=true", OAuth2RequestSQL{Table: sqlTableRefresh}.TableName()),
				id,
				p.NetworkID(ctx),
			).
			Exec(),
	)
```

Reply:
You mean because we test if the config is enabled anyways, right?

Reply:
We're using the correct config when we attempt to use the token, so I don't think we should do it here as well.

Reply:
I don't see why we should not always set the `first_used_at` column.

1. We're updating the row regardless.
1. It can help with debugging to have it set.
2. Enabling graceful refresh later also enables it for recently refreshed tokens, not just for the ones where the setting was already enabled.